pkg/transform/transform.go

```diff
@@ -0,0 +1,158 @@
+package transform
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/fusor/cpma/env"
+	"github.com/fusor/cpma/pkg/io"
+	"github.com/fusor/cpma/pkg/transform/oauth"
+	"github.com/fusor/cpma/pkg/transform/secrets"
+	"github.com/sirupsen/logrus"
+)
+
+// OCP4InstallMsg message about using generated manifests
+const OCP4InstallMsg = `To install OCP4 run the installer as follow in order to add CRDs:
+' /openshift-install --dir $INSTALL_DIR create install-config'
+'./openshift-install --dir $INSTALL_DIR create manifests'
+# Copy generated CRD manifest files  to '$INSTALL_DIR/openshift/'
+# Edit them if needed, then run installation:
+'./openshift-install --dir $INSTALL_DIR  create cluster'`
+const MasterConfigFile = "/etc/origin/master/master-config.yaml"
+const NodeConfigFile = "/etc/origin/node/node-config.yaml"
+const RegistriesConfigFile = "/etc/containers/registries.conf"
+
+type Cluster struct {
+	Master Master
+}
+
+type Master struct {
+	OAuth   oauth.OAuthCRD
+	Secrets []secrets.Secret
+}
+
+type Manifest struct {
+	Name string
+	CRD  []byte
+}
+
+type Config struct {
+	MasterConfigFile     string
+	NodeConfigFile       string
+	RegistriesConfigFile string
+	OutputDir            string
+	Hostname             string
+}
+
+type ManifestTransformOutput struct {
+	//Config    Config
+	Manifests []Manifest
+}
+
+type TransformRunner struct {
+	Config string
+}
+
+// GetFile allows to mock file retrieval
+var GetFile = io.GetFile
+
+func Start() {
+	config := Config{}
```

Comment:
[ISSUE] I think we should put all of the configuration setup before running any of the `Transforms` behind a function, so that `Start` looks like:

```
config := loadConfig(/*possibly inputs*/)
//build and run transforms with config input
```

The end result of "config" will probably continue to become more complex, because we're going to keep adding new things to the config file as we learn more about the problem, and we also need to merge that with the cli provided options, and determine precedence if there is ever a conflict between either of those config sources (do we take the cli option, or the file option? Almost always, cli). Let's abstract that behind a function so we can change that without breaking main.

Reply:
Added a LoadConfig to do this and I think it addresses the concern. If not let me know what more should be done here!